dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -167,3 +165,10 @@ def update(self, df_series):
         self._update_helper(df_series, profile)
 
         return self
+
+    def dict_of_categories(self):
+        """
+        Returns the dict of categories with the number of occurrences
+        for each category.
+        """
+        return dict(self._categories.items())
```

Comment:
Instead of dict_of_categories, we could do a functional property `category_counts` and return a copy of the dict.
However, I suggest adding that to the subsequent PR which adds it to the profile.

Reply:
For now we can just remove this function.